### ISSUE TITLE
Product Layered Navigation filters return incorrect values

### ISSUE THREAD

**Issue by heerdt** [magento/magento2#26372]
The Filters object returned by the Products query should contain the available layered navigation filters array. If the Products query is filtered by category id, that is analogous to the shopper browsing to a category page, so returned GraphQL product filters in layered nav should correspond to the filters shoppers see on the category page.

### Preconditions (*)
Magento 2.3.3
Sample data
Steps to reproduce (*)
Issue GraphQL query

```
{  
   products(filter: {category_id: {eq: "12"}}) {
    total_count
    filters {
      name
      filter_items_count
      filter_items{
        label
        value_string
        items_count
      }
    }
  }
}
```

### Expected result (*)
Returned layered navigation filters are the same as what is shown on the category page, for both search engine implementations: MySQL and ElasticSearch
![image](https://user-images.githubusercontent.com/819013/72258172-4a72df00-3605-11ea-81af-954a7514d2ad.png)

### Actual result (*)
Layered navigation filters do not correspond to what is shown on the category page.
![image](https://user-images.githubusercontent.com/819013/72258257-87d76c80-3605-11ea-9eda-7fb858dccd1d.png)

The returned facets are not using the query/filter applied to return the product list.

This is a blocker to build a proper Search result page or category page using GraphQL.

Thank you

**Comment by heerdt** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @heerdt. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @heerdt, here is your [Magento instance](https://i-26372-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26372-2-4-develop.instances.magento-community.engineering/admin](https://i-26372-2-4-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by heerdt** [comment]
> Hi @heerdt. Thank you for your report.
> To help us process this issue please make sure that you provided the following information:
> 
> * [ ]  Summary of the issue
> * [ ]  Information on your environment
> * [ ]  Steps to reproduce
> * [ ]  Expected and actual results
> 
> Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue:
> 
> `@magento give me 2.4-develop instance` - upcoming 2.4.x release
> 
> For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#contributor-assist) documentation.
> 
> @heerdt do you confirm that you were able to reproduce the issue on vanilla Magento instance following steps to reproduce?
> 
> * [ ]  yes
> * [ ]  no
> 
> * Join [Magento Community Engineering Slack](https://opensource.magento.com/slack) and ask your questions in [#github](https://magentocommeng.slack.com/messages/C7KB93M32) channel.

yes

**Comment by magento-engcom-team** [comment]
Hi @heerdt. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @heerdt, here is your [Magento instance](https://i-26372-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26372-2-4-develop.instances.magento-community.engineering/admin](https://i-26372-2-4-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by heerdt** [comment]
@magento yes, Still reproducible on 2.4-develop

**Comment by engcom-Charlie** [comment]
@heerdt Hello, the field Products.filters is deprecated. Please try to use aggregations instead.

**Comment by engcom-Charlie** [comment]
For example
![image](https://user-images.githubusercontent.com/51680850/72352050-3b199180-36ea-11ea-8a07-9d70a1c5d8b0.png)

**Comment by heerdt** [comment]
hi @engcom-Charlie is this part of the release 2.3.4 or is it going to be release on 2.4? Thanks!

**Comment by lenaorobei** [comment]
Hi @heerdt, as @engcom-Charlie mentioned, aggregations should be used instead. This functionality will be available in `2.3.4` release. For more information please refer to the design document https://github.com/magento/architecture/pull/229

Closing this issue since the described approach is deprecated. 
Thank you.
